internal/client/client_batch.go

```diff
@@ -567,7 +567,14 @@ func (c *batchCommandsClient) isStopped() bool {
 }
 
 func (c *batchCommandsClient) available() int64 {
-	return c.maxConcurrencyRequestLimit.Load() - c.sent.Load()
+	limit := c.maxConcurrencyRequestLimit.Load()
+	sent := c.sent.Load()
+	// just in case, `sent` may less than 0, see https://github.com/tikv/client-go/issues/1225
+	// then maxInt64 - (-1) will overflow, which is unexpected.
+	if sent > 0 {
+		return limit - sent
```

Comment:
Is it acceptable to return a negative result?

Reply:
It's ok here, negative result means not available.

Reply:
Nice cache, it's better not return negative result here.